From c8d4c6ea5a8074688c5934d4480284ba0383e720 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Mon, 25 Dec 2023 19:18:04 -0500
Subject: [PATCH] Enable redefinitions

---
 .SRCINFO | 14 ++++++--------
 PKGBUILD | 47 ++++++++++++++++++++++-------------------------
 2 files changed, 28 insertions(+), 33 deletions(-)

diff --git a/.SRCINFO b/.SRCINFO
index 05ba7c6..b17edd5 100644
--- a/.SRCINFO
+++ b/.SRCINFO
@@ -1,7 +1,7 @@
 pkgbase = cling
 	pkgdesc = Interactive C++ interpreter, built on the top of LLVM and Clang libraries
-	pkgver = 0.9
-	pkgrel = 5
+	pkgver = 1.0
+	pkgrel = 1
 	url = https://root.cern.ch/cling
 	arch = i686
 	arch = x86_64
@@ -15,14 +15,12 @@ pkgbase = cling
 	optdepends = python: support for scan-view and Jupyter
 	optdepends = python-pygments: support for opt-viewer
 	optdepends = python-yaml: support for opt-viewer
-	provides = cling
-	conflicts = cling
 	options = !lto
 	source = cling-llvm::git+http://root.cern/git/llvm.git#tag=cling-v0.9
 	source = cling-clang::git+http://root.cern/git/clang.git#tag=cling-v0.9
-	source = cling::git+http://root.cern/git/cling.git#tag=v0.9
-	sha256sums = SKIP
-	sha256sums = SKIP
-	sha256sums = SKIP
+	source = cling::git+http://root.cern/git/cling.git#tag=v1.0
+	sha512sums = SKIP
+	sha512sums = SKIP
+	sha512sums = SKIP
 
 pkgname = cling
diff --git a/PKGBUILD b/PKGBUILD
index ee1048c..12c96a8 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -1,35 +1,28 @@
 # Maintainer: archibald869 <archibald869 at web dot de>
-
+# Contributor: Carlos Aznarán <caznaranl@uni.pe>
 pkgname=cling
-pkgver=0.9
-pkgrel=5
+pkgver=1.0
+pkgrel=1
 pkgdesc="Interactive C++ interpreter, built on the top of LLVM and Clang libraries"
-arch=("i686" "x86_64")
-url="https://root.cern.ch/cling"
+arch=(i686 x86_64)
+url="https://root.cern.ch/${pkgname}"
 license=("custom:Cling Release License")
-provides=("cling")
-conflicts=("cling")
-depends=("libxml2")
-makedepends=("git" "cmake" "ninja" "python")
-optdepends=(
-    "perl: support for scan-build, ccc-analyzer and c++-analyzer"
-    "python: support for scan-view and Jupyter"
-    "python-pygments: support for opt-viewer"
-    "python-yaml: support for opt-viewer"
-)
-source=(
-    "cling-llvm::git+http://root.cern/git/llvm.git#tag=cling-v$pkgver"
-    "cling-clang::git+http://root.cern/git/clang.git#tag=cling-v$pkgver"
-    "cling::git+http://root.cern/git/cling.git#tag=v$pkgver"
-)
-sha256sums=(
-    "SKIP"
-    "SKIP"
-    "SKIP"
+depends=(libxml2)
+makedepends=(git cmake ninja python)
+optdepends=('perl: support for scan-build, ccc-analyzer and c++-analyzer'
+    'python: support for scan-view and Jupyter'
+    'python-pygments: support for opt-viewer'
+    'python-yaml: support for opt-viewer')
+source=("cling-llvm::git+http://root.cern/git/llvm.git#tag=cling-v0.9"
+    "cling-clang::git+http://root.cern/git/clang.git#tag=cling-v0.9"
+    "cling::git+http://root.cern/git/cling.git#tag=v$pkgver")
+sha512sums=(
+    'SKIP'
+    'SKIP'
+    'SKIP'
 )
 options=('!lto')
 
-
 prepare() {
     if [ ! -h "$srcdir/cling-llvm/tools/clang" ]; then
         ln -s "$srcdir/cling-clang" "$srcdir/cling-llvm/tools/clang"
@@ -42,6 +35,10 @@ prepare() {
     # patch missing header file
     sed -i '/^#include <vector>$/i #include <limits>' \
         "$srcdir/cling-llvm/utils/benchmark/src/benchmark_register.h"
+
+    # https://github.com/root-project/cling/issues/360
+    sed -i 's/AllowRedefinition(0)/AllowRedefinition(1)/' \
+        "$srcdir/cling/include/cling/Interpreter/RuntimeOptions.h"
 }
 
 build() {
-- 
2.43.0

