From c9091f74307023b133f1b87213a33640ef8ab6d4 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n=20Laos?= <caznaranl@uni.pe>
Date: Sat, 22 Jul 2023 13:53:27 -0500
Subject: [PATCH] Enable python bindings

---
 PKGBUILD | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/PKGBUILD b/PKGBUILD
index 13e475c..9b23e20 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ pkgver=9.5.1
 pkgrel=1
 pkgdesc="An Open Source Finite Element Differential Equations Analysis Library"
 arch=("i686" "x86_64")
-url="http://www.dealii.org/"
+url="https://dealii.org"
 license=('LGPL')
 # deal.II depends on Kokkos which is implicitly provided by Trilinos. Trilinos
 # does not yet support configuration with an external copy of Kokkos so do not
@@ -99,6 +99,8 @@ build() {
      cmake_configuration_flags+=" -DLAPACK_INCLUDE_DIRS=${MKLROOT}/include"
   fi
 
+  cmake_configuration_flags+=" -DDEAL_II_WITH_UMFPACK=ON -DDEAL_II_COMPONENT_EXAMPLES=ON -DDEAL_II_COMPONENT_PYTHON_BINDINGS=ON -DDEAL_II_WITH_PETSC=OFF -DDEAL_II_WITH_P4EST=ON"
+
   # For GSL compatibility we need the full link interface, which includes
   # libgslcblas, so disable --as-needed with GCC:
   sed -i '/ENABLE_IF_LINKS(DEAL_II_LINKER_FLAGS "-Wl,--as-needed")/d' \
-- 
2.41.0

