From 11700da48b6f0a0b9523c63e863259398812be98 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Carlos=20Aznar=C3=A1n?= <caznaranl@uni.pe>
Date: Sat, 30 Dec 2023 10:20:35 -0500
Subject: [PATCH] Enable python bindings

---
 PKGBUILD | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/PKGBUILD b/PKGBUILD
index 13e475c..6639203 100644
--- a/PKGBUILD
+++ b/PKGBUILD
@@ -8,7 +8,7 @@ pkgver=9.5.1
 pkgrel=1
 pkgdesc="An Open Source Finite Element Differential Equations Analysis Library"
 arch=("i686" "x86_64")
-url="http://www.dealii.org/"
+url="https://dealii.org"
 license=('LGPL')
 # deal.II depends on Kokkos which is implicitly provided by Trilinos. Trilinos
 # does not yet support configuration with an external copy of Kokkos so do not
@@ -44,8 +44,10 @@ optdepends=(
       )
 makedepends=('cmake')
 install=deal-ii.install
-source=(https://github.com/dealii/dealii/releases/download/v$pkgver/${_realname}-$pkgver.tar.gz)
-sha1sums=('a94989a9e4c3b8599aceacf71d7156621c1d7426')
+source=(https://github.com/dealii/dealii/releases/download/v$pkgver/${_realname}-$pkgver.tar.gz
+https://github.com/dealii/dealii/pull/15856.patch)
+sha1sums=('a94989a9e4c3b8599aceacf71d7156621c1d7426'
+          '594a265dab26aaa99ad109a21ed1d98c19807a41')
 # where to install deal.II: change to something else (e.g., /opt/deal.II/)
 # if desired.
 _installation_prefix=/usr
@@ -70,6 +72,10 @@ build() {
 
   rm -rf "${srcdir}/build"
   mkdir "${srcdir}/build"
+  
+  cd ${srcdir}/${_realname}-$pkgver
+  patch -p1 -i ../15856.patch
+
   cd "${srcdir}/build"
 
   # explicitly disallow bundled packages: this disables bundled copies of boost,
@@ -99,6 +105,8 @@ build() {
      cmake_configuration_flags+=" -DLAPACK_INCLUDE_DIRS=${MKLROOT}/include"
   fi
 
+  cmake_configuration_flags+=" -DDEAL_II_WITH_UMFPACK=ON -DDEAL_II_COMPONENT_EXAMPLES=ON -DDEAL_II_COMPONENT_PYTHON_BINDINGS=ON -DDEAL_II_WITH_PETSC=ON -DDEAL_II_PETSC_WITH_COMPLEX=ON -DDEAL_II_WITH_P4EST=ON -DDEAL_II_WITH_HDF5=ON"
+
   # For GSL compatibility we need the full link interface, which includes
   # libgslcblas, so disable --as-needed with GCC:
   sed -i '/ENABLE_IF_LINKS(DEAL_II_LINKER_FLAGS "-Wl,--as-needed")/d' \
-- 
2.43.0

